src/fontra/core/server.py

```diff
@@ -280,8 +277,15 @@ async def viewPathHandler(
             qs = quote(request.path_qs, safe="")
             raise web.HTTPFound(f"/?ref={qs}")
 
-        path = request.match_info["path"]
-        if not await self.projectManager.projectAvailable(path, authToken):
+        if not request.query:
+            return await self.staticContentHandler(
+                self.viewEntryPoints[viewName], request
+            )
+
+        project = request.query.get("project")
+        if project and not await self.projectManager.projectAvailable(
```

Comment:
Shouldn't this be
```suggestion
        if project is None or not await self.projectManager.projectAvailable(
```
?

Reply:
Yes: I can confirm this changes the case of "we do have a query, but it doesn't have a `project` key" to a 404.